### ISSUE TITLE
query gives 401 www-authenticate API key is missing or invalid for the environment

### ISSUE THREAD

**Issue by sammyjava** [pinecone-io/pinecone-ts-client#25]
I log the index to the console, and it appears to have been initialized. Then I make a query using:
```javascript
    async function getPineconeContexts(index, vector) {
        const queryRequest: QueryRequest = {
            topK: 10,
            vector
        }
        const queryResponse = await index.query({ queryRequest });
        return queryResponse;
    }
```
And the call returns a 401 with the following response headers. (I think the CORS is a red herring, that typically results from 401). The key thing, I think, is `www-authenticate API key is missing or invalid for the environment "us-west1-gcp". Check that the correct environment is specified.` Does the queryRequest payload need to include the API key, somehow?
```
XHROPTIONShttps://pure-gas-8a31658.svc.us-west1-gcp.pinecone.io/query
CORS Missing Allow Origin

	
OPTIONS
	https://pure-gas-8a31658.svc.us-west1-gcp.pinecone.io/query
Status
401
Unauthorized
VersionHTTP/2
Transferred368 B (0 B size)
Referrer Policystrict-origin-when-cross-origin
    content-length
    	114
    date
    	Tue, 14 Mar 2023 23:32:11 GMT
    server
    	envoy
    www-authenticate
    	API key is missing or invalid for the environment "us-west1-gcp". Check that the correct environment is specified.
    X-Firefox-Spdy
    	h2
    	
    Accept
    	*/*
    Accept-Encoding
    	gzip, deflate, br
    Accept-Language
    	en-US,en;q=0.5
    Access-Control-Request-Headers
    	api-key,content-type
    Access-Control-Request-Method
    	POST
    Cache-Control
    	no-cache
    Connection
    	keep-alive
    DNT
    	1
    Host
    	pure-gas-8a31658.svc.us-west1-gcp.pinecone.io
    Origin
    	http://localhost:3000
    Pragma
    	no-cache
    Referer
    	http://localhost:3000/
    Sec-Fetch-Dest
    	empty
    Sec-Fetch-Mode
    	cors
    Sec-Fetch-Site
    	cross-site
    User-Agent
    	Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0
```

**Comment by sammyjava** [comment]
Upon further review, I think an issue is that I cannot form a proper QueryRequest in my React environment.  So if that's the case, we can delete this issue. I'm more interested in seeing the tests run successfuly in a clean clone of this repo. :)

**Comment by sammyjava** [comment]
Ahhh, there is no payload for the request. The QueryRequest isn't being sent to the server! That is bad. I'm not sure the api-key is at fault.

**Comment by rschwabco** [comment]
The code seems perfectly fine. Are you positive there's a vector that's being passed to the function? Other than that I can't really think of anything that would cause this failure.

**Comment by sammyjava** [comment]
Definitely there is a payload, I'm creating it manually. Can you point me to an example that actually works? I have literally not seen one since the tests all fail. Once I see an app that queries an index successfully I'll be all set.

**Comment by rschwabco** [comment]
@sammyjava is this still happening?

**Comment by farrukhjalali** [comment]
I am also facing the same issue. Looks like everything is fine to me. Not sure how to circumvent it?

**Comment by jhamon** [comment]
I'm going to close this issue since there hasn't been recent activity. Today we published a `v1.0.1` release with a ton of improvements. Please try the new release, see the [v1 Migration Guide](https://github.com/pinecone-io/pinecone-ts-client/blob/main/v1-migration.md), and open a new issue if you're still having trouble.
